### DISCUSSION TITLE
Why the test api of gds has double the histories?

### DISCUSSION THREAD

**Discussion by SMorettini** [nasa/fprime#2153] [Q&A] [answered]
In [api.py](https://github.com/fprime-community/fprime-gds/blob/928a264420a1f2b566c4f1a88e7fceae861fa619/src/fprime_gds/common/testing_fw/api.py#L41C1-L44C65) there are the following lines:
```
# these are owned by the GDS and will not be modified by the test API.
self.aggregate_command_history = pipeline.histories.commands
self.aggregate_telemetry_history = pipeline.histories.channels
self.aggregate_event_history = pipeline.histories.events
```
Why do we need to create the history objects for GDS inside the integration tests? From my understanding, the pipeline of GDS and the pipeline for the integration tests are two different pipelines and both contain their history objects. Could someone clarify this point?

I notice that I run my integration tests for hours and the ram usage keeps increasing because I was clearing the test histories but the histories indicated in the code above were not cleaned by GDS. 

P.s.: In my setup, GDS is started separately, am I don't it wrong?

**Comment by LeStarch** [comment, marked as answer]
This is done because the test api is built on top of the GDS's standard code base.  The intention is that those histories get overridden, but that may not be working....

Note: typically one starts the GDS to see the test in the browser (visibilirt) and runs the test API in a terminal to conduct the test.

**Reply by SMorettini** [reply]
I'll add additional comments in the issue directly, thank you

**Comment by LeStarch** [comment]
Making this an issue to look into.
